### PR TITLE
Wait until all pods are running before starting tests

### PR DESCRIPTION
**Pull request by gmarek**
Ref #15139

cc @roberthbailey

### CODE REVIEW

**Comment by wojtek-t** [comment]
e2e failure is a flake:
"Job should run a job to completion when tasks sometimes fail and are locally restarted"

@k8s-bot test this please

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -135,6 +135,31 @@ func cleanupPods(c *client.Client, ns string) {
...
 
+// Waits until all existing pods are scheduled and returns their amount.
+func waitForStableCluster(c *client.Client) int {
+	timeout := 10 * time.Minute
+	startTime := time.Now()
+
+	allPods, err := c.Pods(api.NamespaceAll).List(labels.Everything(), fields.Everything())
+	expectNoError(err)
+	scheduledPods, currentlyNotScheduledPods := getPodsScheduled(allPods)
+	scheduledPodsNo := len(scheduledPods)
```
Comment (wojtek-t) [inline test/e2e/scheduler_predicates.go]: This seems unneded - please remove this variable and just return len(scheduledPods).

**Comment by wojtek-t** [comment]
LGTM - just one minor comments

File: test/e2e/scheduler_predicates.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Done.

**Comment by wojtek-t** [comment]
LGTM

**Pull request merged**